### PR TITLE
Set noImplicitAny to true and fix all errors

### PR DESCRIPTION
**Pull request by sharwell**
⚠️ I am expecting changes will need to be made. Please review the approach I used in each situation carefully to determine if it's what you really want to go with.

Related to #187

### CODE REVIEW

File: src/analysis/analyzer.ts [hunk 1]
```diff
@@ -144,14 +144,15 @@ export class Analyzer extends AnalyzerGen implements vs.Disposable {
...
 		let handler = this.activeRequests[evt.id];
 		let method: string = handler[2];
+		let error: as.RequestError & { method?: string } = evt.error;
 
-		if (evt.error && evt.error.code == "SERVER_ERROR") {
-			evt.error['method'] = method;
-			this.notify(this.requestErrorSubscriptions, <as.RequestError>evt.error);
+		if (error && error.code == "SERVER_ERROR") {
+			error.method = method;
+			this.notify(this.requestErrorSubscriptions, error);
```
Comment (sharwell) [inline src/analysis/analyzer.ts]: :memo: This cast was unnecessary prior to this pull request.

File: src/analysis/analyzer_gen.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import * as as from "./analysis_server_types";
 import * as a from "./analyzer";
 
 export abstract class AnalyzerGen {
-	protected abstract notify<T>(subscriptions: ((notification: T) => void)[], params: T);
+	protected abstract notify<T>(subscriptions: ((notification: T) => void)[], params: T): void;
```
Comment (sharwell) [inline src/analysis/analyzer_gen.ts]: :memo: This appears to be in generated code, but I don't know how to update the original source.

File: src/debug/debug_impl.ts [hunk 3]
```diff
@@ -860,7 +860,7 @@ class ThreadManager {
 
 	removeStoredIds(ids: number[]) {
 		for (let id of ids) {
-			delete this.storeData[id];
```
Comment (sharwell) [inline src/debug/debug_impl.ts]: :memo: I'm pretty sure this was an actual bug (`storeData` vs. `storedData`).

**Comment by DanTup** [comment]
The file `analyzer_gen.ts` is generated from the API docs (https://github.com/Dart-Code/analysis-server-typescript-generator) so shouldn't be manually edited. I'll see if I can get some to go through it this weekend and update it (as well as review all these). Thanks!

File: src/debug/debug_impl.ts (hunk 3 above)
Comment (DanTup) [inline src/debug/debug_impl.ts]: That does look weird, I suspect you're right. I'm not totally familiar with everything in this file as @devoncarew contributed it and it's rather complicated to debug (the irony!). I think Code has incoming changes that might make that easier (debugger attached to multiple processes - the extension and the debugger).

File: src/analysis/analyzer_gen.ts (hunk 2 above)
Comment (DanTup) [inline src/analysis/analyzer_gen.ts]: heh, yep, I commented about that just after. It's a dart script which takes the [API Doc](https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html) and generates all the TS types from it. I don't know if there's enough info for us to do this totally accurately (eg. know what can/can't be null) but certainly changes like this are simple :-)

File: src/debug/debug_impl.ts (hunk 3 above)
Comment (devoncarew) [inline src/debug/debug_impl.ts]: Yup, looks like. I'm guessing this would have manifested mostly as a memory leak?

Comment (DanTup) [inline src/debug/debug_impl.ts]: I suspect so; though how much of how often, I don't know. I'll try and get this merged soon anyway; I've wanted to use these TS options for a while but never got round to completely getting it compiling again!

**Comment by DanTup** [comment]
Actually, the project I linked didn't generate that file at all; I never finished porting it (it only does the types file); it's currently an old crufty C# project not shared! I'll fix it up for now and add porting it to the dart project to my list!

**Comment by DanTup** [comment]
All seems to work fine to me; thanks! 👍

**Pull request merged**